lti_consumer/lti_consumer.py

```diff
@@ -0,0 +1,829 @@
+
+        try:
+            anon_id = parse_handler_suffix(suffix)
+        except LtiError:
+            return Response(status=404)  # 404 because a part of the URL (denoting the anon user id) is invalid
+        try:
+            lti_consumer.verify_result_headers(request, verify_content_type=True)
+        except LtiError:
+            return Response(status=401)  # Unauthorized in this case.  401 is right
+
+        user = self.runtime.get_real_user(anon_id)
+        if not user:  # that means we can't save to database, as we do not have real user id.
+            msg = "[LTI]: Real user not found against anon_id: {}".format(anon_id)
+            log.info(msg)
+            return Response(status=404)  # have to do 404 due to spec, but 400 is better, with error msg in body
+
+        try:
+            # Call the appropriate Lti2Consumer method
+            args = []
+            if request.method == 'PUT':
+                # Request body should be passed as an argument
+                # to result handler method on PUT
+                args.append(request.body)
+            response_body = getattr(lti_consumer, "{}_result".format(request.method.lower()))(user, *args)
+        except (AttributeError, LtiError):
+            return Response(status=404)
+
+        return Response(
+            json.dumps(response_body),
+            content_type=Lti2Consumer.CONTENT_TYPE_RESULT_JSON,
+        )
+
+    def max_score(self):
+        """
+        Returns the configured number of possible points for this component.
+
+        Arguments:
+            None
+
+        Returns:
+            float: The number of possible points for this component
+        """
+        return self.weight if self.has_score else None
+
+    def clear_user_module_score(self, user):
+        """
+        Clears the module user state, including grades and comments, and also scoring in db's courseware_studentmodule
+
+        Arguments:
+            user (django.contrib.auth.models.User):  Actual user whose module state is to be cleared
+
+        Returns:
+            nothing
+        """
+        self.set_user_module_score(user, None, None)
+
+    def set_user_module_score(self, user, score, max_score, comment=u''):
+        """
+        Sets the module user state, including grades and comments, and also scoring in db's courseware_studentmodule
+
+        Arguments:
+            user (django.contrib.auth.models.User):  Actual user whose module state is to be set
+            score (float):  user's numeric score to set.  Must be in the range [0.0, 1.0]
+            max_score (float):  max score that could have been achieved on this module
+            comment (unicode):  comments provided by the grader as feedback to the student
+
+        Returns:
+            nothing
+        """
+        if score is not None and max_score is not None:
+            scaled_score = score * max_score
```

Comment:
I think the score in the published event is the raw score, which is later scaled with the weight automatically.  See https://openedx.atlassian.net/wiki/display/TNL/edX+Grading+Guide+for+Developers to make sure that I remember correctly. :)